bin/configurer/utils/generate-local-user-mappings.sh

```diff
@@ -4,22 +4,50 @@ set -eu
 
 dir=$(dirname ${0})
 root_dir=$(realpath ${dir}/../../..)
-resources_dir=${root_dir}/service/src/main/resources
 
-function get_user_ids() {
-  docker run -e PGPASSWORD='openidm' --rm --network ccd-network postgres:11-alpine psql --host shared-db  --username openidm --tuples-only  --command "SELECT string_agg(fullobject->>'_id',',') FROM managedObjects WHERE fullobject->>'userName' LIKE '%$1%';" openidm
+mock_file=${root_dir}/docker/wiremock/__files/mockResponse.json
+mock_tmp_file=${root_dir}/docker/wiremock/__files/mockResponse.tmp.json
+users_file=${root_dir}/bin/configurer/users.json
+users_ids_tmp_file=${dir}/userIds.json.tmp
+
+user_mappings_file=${root_dir}/service/src/main/resources/application-user-mappings.yaml
+
+function query_db() {
+  docker run -e PGPASSWORD='openidm' --rm --network ccd-network postgres:11-alpine psql --host shared-db  --username openidm --tuples-only  --command "$1" openidm
+}
+
+function get_users_email_id_mappings() {
+  query_db "SELECT json_object(array_agg(fullobject->>'mail'), array_agg(fullobject->>'_id')) FROM managedObjects WHERE fullobject->>'mail' IS NOT NULL AND fullobject->>'userName' LIKE '%$1%';"
+}
+
+function get_users_ids() {
+  query_db "SELECT string_agg(fullobject->>'_id',',') FROM managedObjects WHERE fullobject->>'userName' LIKE '%$1%';"
+}
+
+function create_mock_response() {
+  jq --arg organisation $1 '[.[] | select(.email | contains($organisation))]' $users_file > $mock_file
+  echo $(get_users_email_id_mappings $1) > $users_ids_tmp_file
+
+  for i in `seq 1 $(jq '. | length' $mock_file)`
```

Comment:
Wound't it be easier to start with 0 and not having to do '$index - 1' later?

Reply:
agree :+1: changed